### PR TITLE
Remove namespace concept from Problems API

### PR DESCRIPTION
**Pull request by donat**
Fixes [#30543](https://github.com/gradle/gradle/issues/30543)

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Review by lkasso — COMMENTED** [review]
Docs LGTM.

**Comment by donat** [comment]
@bot-gradle test this

File: platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r86/ProblemProgressEventCrossVersionTest.groovy [hunk 1]
```diff
@@ -97,7 +97,7 @@ class ProblemProgressEventCrossVersionTest extends ToolingApiSpecification {
         given:
         withReportProblemTask """
             for(int i = 0; i < 10; i++) {
-                problems.forNamespace("org.example.plugin").reporting {
+                problems.${targetVersion >= GradleVersion.version("8.11") ? 'getReporter()' : 'forNamespace("org.example.plugin")'}.reporting {
```
Comment (reinsch82) [inline platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r86/ProblemProgressEventCrossVersionTest.groovy:100]: ❌ I think we should extract this and reuse it in the other places in this file.

**Review by reinsch82 — CHANGES_REQUESTED** [review]
Just one refactoring suggested.

File: platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r86/ProblemProgressEventCrossVersionTest.groovy (hunk 1 above)
Comment (donat) [reply platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r86/ProblemProgressEventCrossVersionTest.groovy]: Fixed; ptal

**Review by reinsch82 — APPROVED** [review]
Thx.
LGTM

**Comment by donat** [comment]
@bot-gradle merge

**Pull request merged**